### PR TITLE
InventoryWalk check

### PR DESCRIPTION
**Pull request by UserNugget**
Just to track changes with it and get some feedback

### CODE REVIEW

File: src/main/java/ac/grim/grimac/checks/impl/inventory/InventoryD.java [hunk 1]
```diff
@@ -27,112 +26,88 @@ public InventoryD(GrimPlayer player) {
...
+                return;
+            }
 
-            short lastSentTransaction = (short) (player.lastTransactionSent.get() + 1);
+            controlledMovements++;
 
-            velocityData.add(new Velocity(lastSentTransaction, x, y, z));
-        }
-    }
+            // TODO: close inventory to prevent infinity setbacks
```
Comment (MachineBreaker) [inline src/main/java/ac/grim/grimac/checks/impl/inventory/InventoryD.java]: You could send a close window packet or the bukkit api to do this
Bukkit: `player.bukkitPlayer.closeInventory();`
Packet: 
`sendPacket(new PacketPlayOutCloseWindow(*here the active window id*));`

// drop crafting grid and result slots items
// we should drop carried item?

Comment (UserNugget) [reply src/main/java/ac/grim/grimac/checks/impl/inventory/InventoryD.java]: Fixed by f469b08

File: src/main/java/ac/grim/grimac/events/packets/PacketPlayerWindow.java [hunk 2]
```diff
@@ -93,4 +104,28 @@ public void onPacketSend(PacketSendEvent event) {
                                                 () -> player.hasInventoryOpen = false);
         }
     }
+
+    private boolean isAlwaysDesynced(GrimPlayer player, String legacyType, int modernType) {
+        // Closing beacon with the GUI button causing desync, fixed in 1.9
+        if (player.getClientVersion() == ClientVersion.V_1_8 &&
```
Comment (MachineBreaker) [inline src/main/java/ac/grim/grimac/events/packets/PacketPlayerWindow.java:108]: Is this only on 1.8? or it affects 1.7 as well

Comment (UserNugget) [reply src/main/java/ac/grim/grimac/events/packets/PacketPlayerWindow.java:108]: It affects only 1.8, corrected comment in [9224ab8](https://github.com/MachineBreaker/Grim/pull/1/commits/9224ab8a760b1adfb09c1afdafb924190c738b3c)

**Review by MachineBreaker — APPROVED** [review]
LGTM, is this ready to merge?

**Comment by UserNugget** [comment]
Should be ready now

**Pull request merged**